ccm_web/client/src/utils/validation.ts

```diff
@@ -0,0 +1,46 @@
+import { string, StringSchema, ValidationError } from 'yup'
+
+// Yup: https://github.com/jquense/yup
+
+// Schemas
+
+const createExceededMessage = (fieldName: string, max: number): string => {
+  return `Value for the ${fieldName} must be ${max} characters in length or less.`
+}
+
+const createBlankMessage = (fieldName: string): string => `Value for the ${fieldName} may not be blank.`
+
+const createCanvasEntityNameSchema = (fieldName: string): StringSchema => {
+  return string().required(createBlankMessage(fieldName))
+    .max(255, ({ max }) => createExceededMessage(fieldName, max))
+}
+
+export const courseNameSchema = createCanvasEntityNameSchema('course name')
+export const assignmentHeaderSchema = createCanvasEntityNameSchema('assignment header')
+export const sectionNameSchema = createCanvasEntityNameSchema('section name')
+
+// Type validator(s)
+
+export interface ValidationResult {
+  transformedValue: string | undefined
+  isValid: boolean
+  messages: readonly string[]
+}
+
+export function validateString (value: string | undefined, schema: StringSchema): ValidationResult {
+  let transformedValue: string | undefined
+  let isValid = true
+  let messages: string[] = []
+  try {
+    transformedValue = schema.validateSync(value)
+  } catch (error) {
+    if (!(error instanceof ValidationError)) {
+      throw new Error(String(error))
+    } else {
+      transformedValue = error.value as string | undefined
```

Comment:
I'd rather not use `as` here to coerce the type, but it's basically just as good as what `yup` is provided, which is `any` or `unknown`. I am very confident that the `transformedValue` (which we're not using yet, but could use for trimming white space or changing case) will be of type `string | undefined`, since that is what the wrapper function accepts and passes to `validateSync`.